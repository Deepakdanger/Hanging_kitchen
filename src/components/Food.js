import React from 'react';
import PropTypes from 'prop-types';

const Food = ({ ele }) => {
  const {
    strMeal, strCategory, strArea, strInstructions, strYoutube, strMealThumb,
  } = ele;

  return (
    <div className="food">
      <div className="food_discription">
        <p>
          Food Name :
          {strMeal}
        </p>
        <p>
          Category :
          {strCategory}
        </p>
        <p>
          Area :
          {strArea}
        </p>
        <p>
          Youtube link :
          <a href={strYoutube}>{strYoutube}</a>
        </p>
        <p>
          Instruction :
          {strInstructions}
        </p>
      </div>
      <div className="food_url_discription">
        <p className="food_img">
          <img src={strMealThumb} alt=" " />
        </p>
      </div>
    </div>
  );
};

Food.propTypes = {
  ele: PropTypes.shape({
    strMeal: PropTypes.string,
    strCategory: PropTypes.string,
    strYoutube: PropTypes.string,
    strArea: PropTypes.string,
    strInstructions: PropTypes.string,
    strMealThumb: PropTypes.string,
  }),
};

Food.defaultProps = {
  ele: PropTypes.shape({
    strMeal: '1',
    strCategory: 'fhfh',
    strYoutube: '1',
    strArea: 'fhfh',
    strInstructions: '1',
    strMealThumb: '1',
  }),
};

export default Food;
