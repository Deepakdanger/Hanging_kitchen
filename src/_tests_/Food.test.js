import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { StaticRouter } from 'react-router-dom';
import Food from '../components/Food';

describe('Testing Filter with static router', () => {
  it('Food component is defined', async () => {
    const component = render(
      <StaticRouter>
        <Food ele={{}} />
      </StaticRouter>,
    );
    expect(component).toBeDefined();
  });

  it('Food snapshot testing', () => {
    const food = renderer.create(
      <StaticRouter>
        <Food ele={{}} />
      </StaticRouter>,
    ).toJSON();
    expect(food).toMatchSnapshot();
  });
});
