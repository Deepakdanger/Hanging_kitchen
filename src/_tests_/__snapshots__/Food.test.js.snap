// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Filter with static router Food snapshot testing 1`] = `
<div
  className="food"
>
  <div
    className="food_discription"
  >
    <p>
      Food Name :
    </p>
    <p>
      Category :
    </p>
    <p>
      Area :
    </p>
    <p>
      Youtube link :
      <a />
    </p>
    <p>
      Instruction :
    </p>
  </div>
  <div
    className="food_url_discription"
  >
    <p
      className="food_img"
    >
      <img
        alt=" "
      />
    </p>
  </div>
</div>
`;
